### PR TITLE
Add messages to recurring donations

### PR DESCRIPTION
**Pull request by sampoder**
## Summary of the problem

We had this field but it did nothing!

## Describe your changes

Now values inputted will be saved in the recurring donation record and they should be emailed out. Do we only want to email messages once? I feel like we should and can edit this PR to do that.

### CODE REVIEW

**Comment by sampoder** [comment]
(ignore those schema changes I'll fix them quickly when my laptop gets some juice in it)

**Review by cjdenio — CHANGES_REQUESTED** [review]
Great start! I agree that messages should only be sent for the first donation— you can use `Donation#initial_recurring_donation?` to check

File: app/models/donation.rb [hunk 1]
```diff
@@ -306,7 +310,7 @@ def send_payment_notification_if_needed
 
     if first_donation?
       DonationMailer.with(donation: self).first_donation_notification.deliver_later
-    elsif includes_message?
+    elsif includes_message? && (initial_recurring_donation? || !recurring?)
```
Comment (sampoder) [inline app/models/donation.rb]: @cjdenio that should do this! is there a cleaner way you can think of?

File: app/models/donation.rb
Comment (garyhtou) [inline app/models/donation.rb:1]: Let's remove the changes from this file and include the message here:

https://github.com/hackclub/bank/blob/6ea51d0675f51ed73409a7e185afc5f6f284a816/app/services/recurring_donation_service/handle_invoice_paid.rb#L10-L19

Only include the message if it's the first donation

Comment (garyhtou) [reply app/models/donation.rb:1]: This is because ideally, `Donation` shouldn't know much about `RecurringDonation`.

Comment (garyhtou) [reply app/models/donation.rb:1]: addressed

**Review by garyhtou — APPROVED** [review]
I think this should be good to go now! please sanity check it first (I haven't tested it)

**Pull request merged**